.githooks/pre-commit

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+# Project root for easy directory navigation. In order for our
+# linters/formatters to find their configs, we need to be the in service
+# directory where the config lives.
+PROJ_ROOT="$(pwd)"
+
+# Used as the exit code for the script; if any file fails linting/formatting
+# checks, abort the commit.
+CHECK_FAILED=0
+
+########################
+# Print error message to stderr and set CHECK_FAILED
+# Globals:
+#   CHECK_FAILED
+# Arguments:
+#   Message to display
+########################
+err() {
+  echo "$*" >&2
+  CHECK_FAILED=1
+}
+
+########################
+# Run linting and format checking over a Python file
+# Globals:
+#   PROJ_ROOT
+# Arguments:
+#   File to check, a path
+########################
+check_py() {
+  # FIXME: We should probably determine the service directory dynamically from
+  # the file path instead of hard-coding. We could conceivably end up with
+  # multiple Python services.
+  cd "services/api"
+  git show ":$1" | poetry run flake8 "-qq" "-" || err "[flake8   ]: $1 failed linting"
+  git show ":$1" | poetry run black "--check" "--quiet" "-" || err "[black    ]: $1 needs formatting"
+  cd "${PROJ_ROOT}"
+}
+
+########################
+# Run linting and format checking over a JS, CJS, CSS, SCSS, or Svelte file.
+# Globals:
+#   PROJ_ROOT
+# Arguments:
+#   File to check, a path
+#   Type of file being checked (js, cjs, svelte, css, or scss)
+########################
+check_web() {
+  # FIXME: We should probably determine the service directory dynamically from
+  # the file path instead of hard-coding. We could conceivably end up with
+  # multiple Python services.
```

Comment:
A small nit - the `check_web()` function works on node/js services, and not Python ones.

```suggestion
  # multiple Node services.
```